src/open_inwoner/kvk/tests/test_views.py

```diff
@@ -101,13 +157,18 @@ def test_post_branches_page_with_incorrect_vestigingsnummer(
         mock_solo.return_value.client_certificate = CertificateFactory()
         mock_solo.return_value.server_certificate = CertificateFactory()
 
-        self.client.force_login(user=self.user)
-
+        self.client.force_login(
+            user=self.user,
+            backend=EHerkenningSessionContext._expected_auth_backends()[0],
+        )
         response = self.client.post(self.url, data={"branch_number": "4321"})
 
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(kvk_branch_selected_done(self.client.session), False)
-        self.assertNotIn(KVK_BRANCH_SESSION_VARIABLE, self.client.session)
+        self.assertEqual(
+            response.wsgi_request.user,
+            self.user,
+            msg="No branch change effectuated",
```

Comment:
`No branch change effectuated` is a bit cryptic/elaborate. Perhaps `KVK branch was not changed`